components/blitz/src/omero/cmd/graphs/GraphHelper.java

```diff
@@ -51,15 +56,34 @@
 
     private final Helper helper;
     private final GraphPathBean graphPathBean;
+    private final LightAdminPrivileges adminPrivileges;
 
     /**
      * Construct a helper for a graph request instance.
      * @param helper the general request helper for the graph request instance
      * @param graphPathBean the graph path bean
+     * @param adminPrivileges the light administrator privileges helper
      */
-    public GraphHelper(Helper helper, GraphPathBean graphPathBean) {
+    public GraphHelper(Helper helper, GraphPathBean graphPathBean, LightAdminPrivileges adminPrivileges) {
         this.helper = helper;
         this.graphPathBean = graphPathBean;
+        this.adminPrivileges = adminPrivileges;
+    }
+
+    /**
+     * Check if the current user is an administrator.
+     * @param requiredPrivilege the privilege that the administrator must have if they are a light administrator
+     * @return if the current user is an administrator
+     */
+    public boolean checkIsAdministrator(AdminPrivilege requiredPrivilege) {
+        final EventContext eventContext = helper.getEventContext();
+        if (!eventContext.isCurrentUserAdmin()) {
```

Comment:
Yes, one thing I got from the "restrictions" discussion was that light admins are members of `system`. I did experiment with trying to introduce a string set of privileges in event context but got confused over interactions with sudo and intimidated by fear of getting things wrong. My current thinking is to try to make new roles minimally invasive for 5.3 then once we have more time and experience to fit them in more naturally for 5.4 and your suggestion is a prime candidate for how that needs to be done.